src/libsyntax/diagnostic_list.rs

```diff
@@ -16,31 +16,394 @@
+E0539: r##"
+A named attribute parameter doesn't have a value.
+
+Erroneous code example:
+
+```compile_fail
+#[stable(feature)] // error: incorrect meta item
+pub fn something() {}
+```
+
+Meta items need a value:
+
+```ignore
+#[stable(feature = "some_feature")] // error: incorrect meta item
+pub fn something() {}
+```
+"##,
+
+E0540: r##"
+Multiple conflicting rustc_deprecated annotations (the attribute) for the same
+item (function/static/etc...).
+
+Erroneous code example:
+
+```compile_fail
+#[rustc_deprecated(since = "1.3.0", reason = "some reason")]
+#[rustc_deprecated(since = "1.4.0", reason = "no reason")]
+// error: multiple rustc_deprecated attributes
+pub fn something() {}
+```
+
+For obvious reason, an element can only be deprecated once. Example:
+
+```ignore
+#[rustc_deprecated(since = "1.3.0", reason = "some reason")] // ok!
+pub fn something() {}
+```
+"##,
+
+E0541: r##"
+An unknown named attribute parameter was used on rustc_deprecated attribute.
+
+Erroneous code example:
+
+```compile_fail
+#[rustc_deprecated(since = "1.3.0", reason = "some reason", some = "unknown")]
+// error: unknown meta item
+pub fn something() {}
+```
+
+The rustc_deprecated attribute only recognizes the two following named
+attribute parameters:
+
+ * since
+ * reason
+
+Example:
+
+```ignore
+#[rustc_deprecated(since = "1.3.0", reason = "some reason")] // ok!
+pub fn something() {}
+```
+"##,
+
+E0542: r##"
+The "since" named attribute parameter wasn't used in rustc_deprecated
+attribute.
+
+Erroneous code example:
+
+```compile_fail
+#[rustc_deprecated(reason = "some reason")] // error: missing 'since'
+pub fn something() {}
+```
+
+The two following named attribute parameters must be given to the
+rustc_deprecated attribute:
+
+ * since
+ * reason
+
+Example:
+
+```ignore
+#[rustc_deprecated(since = "1.3.0", reason = "some reason")] // ok!
+pub fn something() {}
+```
+"##,
+
+E0543: r##"
+The "reason" named attribute parameter wasn't used in rustc_deprecated
+attribute.
+
+Erroneous code example:
+
+```compile_fail
+#[rustc_deprecated(since = "1.3.0")] // error: missing 'reason'
+pub fn something() {}
+```
+
+The two following named attribute parameters must be given to the
+rustc_deprecated attribute:
+
+ * since
+ * reason
+
+Example:
+
+```ignore
+#[rustc_deprecated(since = "1.3.0", reason = "some reason")] // ok!
+pub fn something() {}
+```
+"##,
+
+E0544: r##"
+Conflicting stability levels applied to an item.
+
+Erroneous code example:
+
+```compile_fail
+#[stable(feature = "some_stable_feature")]
+#[unstable(feature = "some_unstable_feature", issue = "1")]
+// error: multiple stability levels
+pub fn something() {}
+```
+
+An element can only have one stability level. Example:
+
+```ignore
+#[unstable(feature = "some_unstable_feature", issue = "1")] // ok!
+pub fn something() {}
+```
+
+Or:
+
+```ignore
+#[stable(feature = "some_stable_feature")] // ok!
+pub fn something() {}
+```
+
+However, an element can be stable and deprecated. Example:
```

Comment:
It can also be unstable and deprecated, so perhaps a “However, an item can be deprecated regardless of its stability level. Example:” would be better.